ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/phase0/util/AttestationUtilPhase0.java

```diff
@@ -34,4 +44,70 @@ public AttestationUtilPhase0(
   public AttestationWorthinessChecker createAttestationWorthinessChecker(final BeaconState state) {
     return AttestationWorthinessChecker.NOOP;
   }
+
+  /**
+   * [IGNORE] aggregate.data.slot is within the last ATTESTATION_PROPAGATION_SLOT_RANGE slots (with
+   * a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. aggregate.data.slot +
+   * ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= aggregate.data.slot (a client MAY queue
+   * future aggregates for processing at the appropriate slot).
+   */
+  @Override
+  public Optional<SlotInclusionGossipValidationResult> performSlotInclusionGossipValidation(
+      final Attestation attestation, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
+    final UInt64 attestationSlot = attestation.getData().getSlot();
+    if (isCurrentTimeAfterAttestationPropagationSlotRange(
+            attestationSlot, genesisTime, currentTimeMillis)
+        || isFromFarFuture(attestation, genesisTime, currentTimeMillis)) {
+      return Optional.of(SlotInclusionGossipValidationResult.IGNORE);
+    }
+    if (isCurrentTimeBeforeMinimumAttestationBroadcastTime(
+        attestationSlot, genesisTime, currentTimeMillis)) {
+      return Optional.of(SlotInclusionGossipValidationResult.SAVE_FOR_FUTURE);
+    }
+    return Optional.empty();
+  }
+
+  protected boolean isCurrentTimeBeforeMinimumAttestationBroadcastTime(
```

Comment:
do we need this to be `protected`?

Reply:
I am using the same method in the Deneb version